### PR TITLE
Connect callback

### PR DESCRIPTION
**Pull request by martinsix**
We were calling the `onConnected` callback before finishing setting up
the connection.

### CODE REVIEW

File: src/ConnectionAutomaton.cpp [hunk 1]
```diff
@@ -87,6 +86,7 @@ void ConnectionAutomaton::connect(
       keepaliveTimer_->start(shared_from_this());
     }
   }
+  onConnected_();
```
Comment (lehecka) [inline src/ConnectionAutomaton.cpp]: by changing this order, we might get into the situation that sending frames on line 82 may terminate socket and the client code would get onClose before onConnected.

**Comment by lehecka** [comment]
this needs a different solution. Will rest this change for now.

**Comment by lexs** [comment]
I have some ideas to "fix" onConnected once and for all but that requires delegating the behavior to DuplexConnection, which is a larger change (and I won't get that done in the near future).

**Comment by lehecka** [comment]
closing as obsolete.

**Pull request closed without merging**